framework/test/bundles/libH/TestBundleH.cpp

```diff
@@ -187,6 +168,50 @@ class TestBundleHServiceFactoryReturnsNullPtr : public ServiceFactory
 
 };
 
+// Simulate an error condition whereby the service factory recursively tries to get the same service for the same bundle
+class TestBundleHServiceFactoryRecursion : public ServiceFactory
+{
+public:
+
+  InterfaceMapConstPtr GetService(const Bundle& caller, const ServiceRegistrationBase& sReg)
+  {
+    static bool validRecursion = false;
```

Comment:
Okay, yes the return value of `GetService` is different because of these two test cases being covered. Your snippet is much nicer, I will use it!